wrappers/dotnet/src/Hyperledger.Ursa.BbsSignatures/Keys/BlsKeyPair.cs

```diff
@@ -70,24 +77,44 @@ public BlsKeyPair(byte[] keyData)
         /// </summary>
         /// <param name="messageCount">The message count.</param>
         /// <returns></returns>
-        public BbsKeyPair GeyBbsKeyPair(uint messageCount)
+        public BbsKey GetBbsKey(uint messageCount)
         {
-            using var context = new UnmanagedMemoryContext();
+            using var context = new UnmanagedMemory();
 
             if (SecretKey is null)
             {
                 NativeMethods.bls_public_key_to_bbs_key(context.ToBuffer(PublicKey), messageCount, out var publicKey, out var error);
                 context.ThrowOnError(error);
 
-                return new BbsKeyPair(context.ToByteArray(publicKey), messageCount);
+                return new BbsKey(context.ToByteArray(publicKey), messageCount);
             }
             else
             {
                 NativeMethods.bls_secret_key_to_bbs_key(context.ToBuffer(SecretKey), messageCount, out var publicKey, out var error);
                 context.ThrowOnError(error);
 
-                return new BbsKeyPair(context.ToByteArray(publicKey), messageCount);
+                return new BbsKey(context.ToByteArray(publicKey), messageCount);
             }
         }
+
+        /// <summary>
+        /// Creates new <see cref="BlsKeyPair"/> using a input seed as byte array.
+        /// </summary>
+        /// <param name="seed">The seed.</param>
+        /// <returns></returns>
+        public static BlsKeyPair Generate(string? seed = null)
```

Comment:
Just for context in this library there will soon be support for BLS12-381 G1 and G2 key gen support so if you want to account for that in this API now then take a look at the [objective-c](https://github.com/mattrglobal/ffi-bbs-signatures/pull/15) wrapper

Reply:
I was thinking about adding a field to denote the specific curve in the `BlsKeyPair` class, which should become `record` once net5 comes out. You think that's a good approach?

Reply:
Yeap I think both is worth doing

1. Add a member to the BLSKeyPair that denotes the field G1 or G2 of a generated key
2. Either add a parameter to the generate function to require what field you want to generate the key for or have seperate functions for the different key pairs [like we have done here](https://github.com/mattrglobal/node-bbs-signatures/blob/master/src/bls12381.ts#L60)